src/main/java/org/broadinstitute/hellbender/tools/exome/HetPulldownCalculator.java

```diff
@@ -52,10 +56,13 @@
      */
     public HetPulldownCalculator(final File refFile, final File snpFile,
                                  final int minMappingQuality, final int minBaseQuality) {
+        ParamUtils.isPositiveOrZero(minMappingQuality, "Minimum mapping quality must be nonnegative.");
+        ParamUtils.isPositiveOrZero(minBaseQuality, "Minimum base quality must be nonnegative.");
         this.refFile = refFile;
         this.snpIntervals = IntervalList.fromFile(snpFile);
         this.minMappingQuality = minMappingQuality;
         this.minBaseQuality = minBaseQuality;
+        this.samReaderFactory = SamReaderFactory.makeDefault();     //will use default validation stringency
```

Comment:
Rather than caching the reader factory derived from setting the global state, see the suggested code fragment below.